source/common/http/conn_manager_impl.cc

```diff
@@ -1092,6 +1136,31 @@ void ConnectionManagerImpl::ActiveStreamEncoderFilter::
 
 void ConnectionManagerImpl::ActiveStreamEncoderFilter::continueEncoding() { commonContinue(); }
 
+void ConnectionManagerImpl::ActiveStreamEncoderFilter::responseDataTooLarge() {
+  if (parent_.encoder_filters_all_streaming_) {
+    onEncoderFilterAboveWriteBufferHighWatermark();
+  } else {
+    // FIXME 500
+    HeaderMapPtr response_headers{new HeaderMapImpl{
```

Comment:
I guess one could argue this is filter bug. Not sure. (Like if filter continues headers but then buffers data it's really a streaming filter). Anyway needs a bit of thinking IMO.